splice_machine/src/test/java/com/splicemachine/benchmark/execution/InsertBenchmark.java

```diff
@@ -0,0 +1,175 @@
+import java.util.Arrays;
+import java.util.Collection;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@Category(Benchmark.class)
+@RunWith(Parameterized.class)
+public class InsertBenchmark extends ExecutionBenchmark {
+    private static final String SCHEMA = InsertBenchmark.class.getSimpleName();
+
+    private final int numTableColumns;
+    private final int numIndexTables;
+    private final int dataSize;
+    private final boolean onOlap;
+
+    private final String tableDefStr;
+    private final String indexDefStr;
+    private final String insertParamStr;
+
+    @ClassRule
+    public static SpliceSchemaWatcher spliceSchemaWatcher = new SpliceSchemaWatcher(SCHEMA);
+
+    public InsertBenchmark(int numTableColumns, int numIndexColumns, int numIndexTables, int dataSize, boolean onOlap) {
+        this.numTableColumns = Math.max(numTableColumns, 1);
+        this.numIndexTables = numIndexTables;
+        this.dataSize = dataSize;
+        this.onOlap = onOlap;
+
+        int realIndexColumns = Math.max(Math.min(numIndexColumns, numTableColumns), 1);
+
+        this.tableDefStr = getColumnDef(this.numTableColumns, TABLE_DEF);
+        this.indexDefStr = getColumnDef(realIndexColumns, INDEX_DEF);
+        this.insertParamStr = getColumnDef(this.numTableColumns, INSERT_PARAM);
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        getInfo();
+
+        LOG.info("Setup connection and creating tables");
+        testConnection = makeConnection(spliceSchemaWatcher);
+        testStatement = testConnection.createStatement();
+        testStatement.execute("CREATE TABLE " + BASE_TABLE + tableDefStr);
+
+        createIndexes(testStatement, numIndexTables, indexDefStr);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        spliceSchemaWatcher.cleanSchemaObjects();
+        testStatement.close();
+        testConnection.close();
+        reportStats();
+    }
+
+    private void benchmark(boolean isBatch) {
+        try {
+            // warmup
+            if (isBatch)
+                runBatchInserts(true, curSize.getAndAdd(dataSize));
+            else
+                runInserts(true, curSize.getAndAdd(dataSize));
+
+            if (isBatch)
+                runBatchInserts(false, curSize.getAndAdd(dataSize));
+            else
+                runInserts(false, curSize.getAndAdd(dataSize));
+        } catch (SQLException e) {
+            LOG.error("Error occurs while initializing test connection", e);
+        }
+    }
+
+    private void runInserts(boolean warmup, int concurrency) throws SQLException {
+        try (PreparedStatement insert1 = testConnection.prepareStatement("INSERT INTO " + BASE_TABLE + insertParamStr)) {
+            for (int i = 0; i < ExecutionBenchmark.NUM_WARMUP_RUNS; ++i) {
```

Comment:
It would seem logical to use _NUM_WARMUP_RUNS_ when _warmup_ == _true_ and _NUM_EXECS_ otherwise.